### PR TITLE
PROD-734 #2988 Fixed mentioning in blog post comments doesn't send notifications

### LINKED ISSUE #2988: Mentioning in blog post comments doesn't send notification
**Describe the bug**
Users do not receive any notification if they're being mentioned in a blog post

**To Reproduce**
1. Go to blog
2. Open any blog
3. Comment and mention any user
4. View as the user you tagged
5. See error 

**Screenshots**
https://www.awesomescreenshot.com/video/5420202?key=89a831cc67cfd7b4fba9c7a7ffbef1c7

**Environment**
BuddyBoss Demo

**Support ticket links**
https://secure.helpscout.net/conversation/1611635617/160598/

**Jira issue** : [PROD-734]

[PROD-734]: https://buddyboss.atlassian.net/browse/PROD-734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by bb-yudhisthir**
### All Submissions:
https://buddyboss.atlassian.net/browse/PROD-5126

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2988.

### How to test the changes in this Pull Request:

1. Enable the wordpress post and comments settings in the buddyboss settings > activities under "Posts in Activity Feeds"
2. Create a new sample blog
3. Comment and mention any user in the blog comment
4. View as the user you tagged, Notification should be there as well as an Email for the mention

### Proof Screenshots or Video

http://somup.com/cr6UXi3EJr

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Mentioning in blog post comments sends notifications

### CODE REVIEW

File: src/bp-activity/bp-activity-filters.php [hunk 1]
```diff
@@ -160,6 +160,96 @@
 // Filter check content empty or not for the media, document and GIF data.
 add_filter( 'bb_is_activity_content_empty', 'bb_check_is_activity_content_empty' );
 
+//Action notification for mentions in single page blog comments
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:163]:
```suggestion
// Action notification for mentions in single page blog comments
```

File: src/bp-activity/bp-activity-filters.php [hunk 2]
```diff
@@ -160,6 +160,96 @@
...
+ * Notification for mentions in blog comment
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param int        $activity_id          The activity comment ID.
+ * @param WP_Comment $post_type_comment    WP Comment object.
+ * @param array      $activity_args        Activity comment arguments.
+ * @param object     $activity_post_object The post type tracking args object.
+ */
+function bb_blogs_comment_mention_notification( $activity_id, $comment, $activity_args, $activity_post_object ){
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:176]:
```suggestion
function bb_blogs_comment_mention_notification( $activity_id, $comment, $activity_args, $activity_post_object ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need PHPCS changes

**Review by dipskakadiya — DISMISSED** [review]

File: src/bp-activity/bp-activity-filters.php [hunk 3]
```diff
@@ -160,6 +160,96 @@
...
+				$notifications = BP_Notifications_Notification::get_all_for_user( $user_id, 'all' );
+
+				// Don't leave multiple notifications for the same activity item.
+				foreach ( $notifications as $notification ) {
+					if ( $activity_id == $notification->item_id && 'new_at_mention' === $notification->component_action ) {
+						return;
+					}
+				}
+				// Now email the user with the contents of the message (if they have enabled email notifications).
+				if ( 'no' != bp_get_user_meta( $user_id, 'notification_activity_new_mention', true ) ) {
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:220]:
```suggestion
				if ( 'no' !== bp_get_user_meta( $user_id, 'notification_activity_new_mention', true ) ) {
```

File: src/bp-activity/bp-activity-filters.php [hunk 4]
```diff
@@ -160,6 +160,96 @@
...
+				$email_type   = 'activity-at-message';
+				$group_name   = '';
+				$message_link = bp_activity_get_permalink( $activity_id );
+				$poster_name  = bp_core_get_user_displayname( $activity->user_id );
+
+				$notifications = BP_Notifications_Notification::get_all_for_user( $user_id, 'all' );
+
+				// Don't leave multiple notifications for the same activity item.
+				foreach ( $notifications as $notification ) {
+					if ( $activity_id == $notification->item_id && 'new_at_mention' === $notification->component_action ) {
```
Comment (chetansatasiya) [inline src/bp-activity/bp-activity-filters.php:215]:
```suggestion
					if ( $activity_id === $notification->item_id && 'new_at_mention' === $notification->component_action ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need changes

**Review by chetansatasiya — APPROVED** [review]

**Comment by joeytoribio** [comment]
Hi @bb-yudhisthir,

This has been tested in the qa-ongoing branch without any issues.

### Test Case#1
Mentioning in blog post comments doesn't send notification - Push Notification

**To Reproduce**
1] Enable the WordPress post and comments settings in the Buddyboss settings > activity under "Posts in Activity Feeds"
2] Create a new sample blog
3] Comment and mention any user in the blog comment
4] Make sure that the mentioned user is login in a separate browser to verify the web notification

**Remarks:** The User receives a web push notification

**Result: PASSED**

### Test Case#2
Mentioning in blog post comments doesn't send notification - Email

**To Reproduce**
1] Enable the WordPress post and comments settings in the buddyboss settings > activity under ""Posts in Activity Feeds""
2] Create a new sample blog
3] Comment and mention any user in the blog comment
4] Open tagged user's email to verify

**Remarks:** The user receives an email notification

**Result: PASSED**

### Test Case#3
Mentioning in blog post comments doesn't send notification - Notification Tab

**To Reproduce**
1] Enable the WordPress post and comments settings in the buddyboss settings > activity under "Posts in Activity Feeds"
2] Create a new sample blog
3] Comment and mention any user in the blog comment
4] View as the user you tagged and go to Notification Icon

**Remarks:** The user receives a notification in the notification bell icon

**Result: PASSED**

**Video:** 
https://www.loom.com/share/918581a3b8ef46ccadfeb20e56310d42

Thank you!

**Comment by basil-buddyboss** [comment]
Hi @bb-yudhisthir 

This has been tested in the QA-approved branch including the three test cases and there were no issues encountered.

Result: Passed

**Comment by sourabhmatolia** [comment]
**UPDATE**

This is not done correctly. I will share more details later. Thanks.

**Review by jitendrabanjara1991 — DISMISSED** [review]

**Pull request merged**